### PR TITLE
[charts/redis-ha] Add kubernetes lifecycle hook

### PR DESCRIPTION
**Pull request by nublikasa**
#### What this PR does / why we need it:
1) Read .Values.haproxy.customConfig as a template to be consistent with other customConfigs
2) Add Kubernetes lifecycle hook values. The lifecycle hook especially preStop is very useful during cluster upgrade where haproxy can be gracefully shutdown and redis failover can be invoked manually to prevent connection failure.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by nublikasa** [comment]
@DandyDeveloper Just tagging you here for review. Thanks!

File: charts/redis-ha/values.yaml [hunk 1]
```diff
@@ -120,6 +120,7 @@ haproxy:
   ## Place any additional configuration section to add to the default config-haproxy.cfg
   # extraConfig: |-
       # Define configuration here
```
Comment (DandyDeveloper) [inline charts/redis-ha/values.yaml:122]: Can you fix the indentation here?

File: charts/redis-ha/values.yaml [hunk 2]
```diff
@@ -175,6 +176,8 @@ redis:
   #  limits:
   #    memory: 700Mi
 
+  lifecycle: {}
```
Comment (DandyDeveloper) [inline charts/redis-ha/values.yaml:179]: Can you provide a link to people who potentially don't understand the lifecycle like: 

> Pod lifecycle command for cases of shutdown... etc etc https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/

**Comment by DandyDeveloper** [comment]
@nublikasa Lets do 4.5.10!

**Comment by nublikasa** [comment]
> @nublikasa Lets do 4.5.10!

Done as well 😃

**Comment by nublikasa** [comment]
> @nublikasa Can you do this post pulling from the most recent master? Need to fix the conflicts.

I just did a rebase.

**Comment by DandyDeveloper** [comment]
@nublikasa Looks like you've got a trailing space? CI is failing. Can you sort that? Thank you!

**Comment by nublikasa** [comment]
@DandyDeveloper All good now :)

**Pull request merged**